src/storage/src/hummock/state_store.rs

```diff
@@ -122,25 +122,25 @@ impl StateStoreRead for HummockStorage {
         epoch: u64,
         read_options: ReadOptions,
     ) -> Self::IterFuture<'_> {
-        if let Some(prefix_hint) = read_options.prefix_hint.as_ref() {
-            let next_key = next_key(prefix_hint);
+        if let Some(dist_key_hint) = read_options.dist_key_hint.as_ref() {
```

Comment:
Please correct me if I am wrong @st1page, streaming will still pass the full prefix range bound to stroage when the dist_key is no longer pk prefix. cc 

In this case, the range bouund assertion checks here need to be modified instead of just renaming if `dist_key_hint` is not the prefix of the bound.